examples/some-issuer/frontend/src/shared/util.ts

```diff
@@ -0,0 +1,121 @@
+
+interface IssuerResponse {
+    txHash: string;
+    credential: {
+        proof: CredentialProof;
+        randomness: Record<string, string>;
+    };
+}
+
+interface IssueRequest {
+    credential: CredentialInfo;
+    telegramUser?: TelegramUser;
+}
+
+interface CredentialInfo {
+    holderId: string;
+    holderRevocable: boolean;
+    validFrom: string;
+    metadataUrl: {
+        url: string;
+    };
+}
+
+interface TelegramRequest {
+    platform: Platform.Telegram;
+    user: MakeRequired<TelegramUser, 'username'>;
+}
+
+interface DiscordUser {
+    id: string;
+    username: string;
+}
+
+interface DiscordRequest {
+    platform: Platform.Discord;
+    user: DiscordUser;
+}
+
+interface RpcError {
+    code: string;
+}
+
+const CONTRACT_DID = `$did:ccd:${config.network}:sci:${config.contract.index}:${config.contract.subindex}/issuer`;
+const ISSUER_URL = location.href;
+
+export async function requestCredential(req: TelegramRequest | DiscordRequest) {
+    const { id, username } = req.user;
+
+    const credential = {
+        $schema: `./JsonSchema2023-${req.platform}.json`,
+        type: ['VerifiableCredential', 'ConcordiumVerifiableCredential', 'SoMeCredential'],
+        issuer: CONTRACT_DID,
+        issuanceDate: new Date().toISOString(),
+        credentialSubject: { attributes: { userId: id.toString(), username } },
+        credentialSchema: {
+            id: `${ISSUER_URL}json-schemas/JsonSchema2023-${req.platform}.json`,
+            type: 'JsonSchema2023',
+        },
+    };
+
+    const metadataUrl = {
+        url: ISSUER_URL + 'json-schemas/credential-metadata.json',
+    };
+
+    let txHash: string | undefined;
+    const api = await connectWallet();
+    await api.addWeb3IdCredential(credential, metadataUrl, async (id) => {
+        const parts = id.split(':');
+        const holderId = parts[parts.length - 1];
+
+        const body: IssueRequest = {
+            credential: {
+                holderId,
+                holderRevocable: true,
+                validFrom: new Date().toISOString(),
+                metadataUrl,
+            },
+        };
+        if (req.platform === Platform.Telegram) body.telegramUser = req.user;
+
+        const endpoint = ISSUER_URL + 'credential';
+        const response = await fetch(endpoint, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            // If we're sending Discord requests, we need to include proof of our user id
+            credentials: 'include',
+            body: JSON.stringify(body),
+        });
+
+        if (!response.ok) throw new Error('Error getting credential: ' + (await response.text()));
+
+        const { txHash: hash, credential } = (await response.json()) as IssuerResponse;
+        txHash = hash;
+        const { proof, randomness } = credential;
+        return { proof, randomness };
+    });
+
+    console.log('Transaction submitted, hash:', txHash);
```

Comment:
Do we want to keep this?

Reply:
Well.. it's good for debugging, and it doesn't hurt much does it?

Reply:
I think it does. Any time I open a console on the page and see all kinds of debug logs I think people don't care about quality.

Reply:
Could at least do `console.debug` and then you only see it if you care to enable `verbose` mode.

Reply:
I added something that provides better user feedback instead.